### ISSUE TITLE
Solargraph formatting doesn't follow the project's .rubocop.yml file

### ISSUE THREAD

**Issue by guilhermedeandrade** [castwide/vscode-solargraph#206]
Solargraph formatting doesn't follow the project's `.rubocop.yml` file.

Here is my `.solargraph.yml` file:

```yaml
---
include:
  - "**/*.rb"
exclude:
  - spec/**/*
  - test/**/*
  - vendor/**/*
  - ".bundle/**/*"
require: []
domains: []
reporters:
  - rubocop
  - require_not_found
require_paths: []
plugins: []
max_files: 5000
```

And here is a part of `settings.json`:

```json
{
 ...
  "solargraph.diagnostics": true,
  "solargraph.hover": true,
  "solargraph.formatting": true
}
```

Example file:

```rb
class Cat
  def meow
    'meow!'
  end
end
```

Even though `.rubocop.yml` contains the following cop:
```yaml
Style/StringLiterals:
  EnforcedStyle: double_quotes
```

Solargraph won't format the `'meow!'` to `"meow"!`. This is just an example, the same thing happens with others cops.

Also, somehow, if I do write the class like this:
```rb
class Cat
  def meow
    "meow!"
  end
end
```

The Solargraph formatter will "work" and replace the double quotes for single quotes.

**Comment by dim** [comment]
Same here!

**Comment by jaredcwhite** [comment]
Same here…and in fact another project where it had been working for quite some time is now not formatting anything at all after upgrading to latest Solargraph version. Just silently fails to do anything at all. :(

**Comment by jaredcwhite** [comment]
For the time being, I've turned off SolarGraph's formatter and I'm using the File Watcher extension to run RuboCop after save. FWIW, here's my settings.json:

```json
  "filewatcher.commands": [
    {
      "match": "\\.rb",
      "isAsync": true,
      "cmd": "/bin/zsh --login -c \"cd ${fileDirname}; rubocop --force-exclusion -a ${file}\"",
      "event": "onFileChange"
    }  
  ]
```

**Comment by castwide** [comment]
This problem should be fixed in gem version 0.40.3. If the problem persists, please reply in the related issue in the gem repo: https://github.com/castwide/solargraph/issues/406
